### PR TITLE
[C-2644] Fix issue where new playlist drag-drop not persisting

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Fixes issue where drag-dropping new (unfavorited) playlists into the left nav wouldnt appear. Also fixes case of drag-dropping to specific sections in the left nav, including inside folders
- Refactors playlist-library saga to use `typed-redux-saga` which caught a few type bugs.
- Small styling fix with overflowing text. Now that we wrap everything in a span, we also need to text wrap it.

### CODE REVIEW

File: packages/web/src/components/nav/desktop/PlaylistLibrary/CollectionNavItem.module.css [hunk 1]
```diff
@@ -9,6 +9,7 @@
   display: flex;
   align-items: center;
   line-height: normal;
+  overflow: hidden;
```
Comment (Kyle-Shanks) [inline packages/web/src/components/nav/desktop/PlaylistLibrary/CollectionNavItem.module.css:12]: why was this needed? just curious

Comment (dylanjeffers) [reply packages/web/src/components/nav/desktop/PlaylistLibrary/CollectionNavItem.module.css:12]: oh i see your comment now :) yes as discussed not needed, thanks!
